src/main/java/com/szeloch/jan/parkingsystemmanagement/rest/controller/parkingclient/ParkingClientsController.java

```diff
@@ -0,0 +1,192 @@
+                        ControllerLinkBuilder.linkTo(methodOn(ParkingClientsController.class).findParkingTicketForVehicle(id, vehicle.getId())).withRel("parking-ticket"),
+                        ControllerLinkBuilder.linkTo(methodOn(ParkingClientsController.class).activateParkingTicket(id, vehicle.getId())).withRel("activate-parking-ticket"),
+                        ControllerLinkBuilder.linkTo(methodOn(ParkingClientsController.class).closeParkingTicket(id, vehicle.getId())).withRel("close-parking-ticket")))
+                .collect(Collectors.toList());
+
+        return new Resources<>(parkingClientsVehicles,
+                linkTo(methodOn(VehiclesController.class).findAll()).withRel("vehicles"));
+    }
+
+    @GetMapping("/{clientId}/vehicles/{vehicle-id}/parking-ticket")
+    public ResponseEntity<ParkingTicketResource> findParkingTicketForVehicle(@PathVariable("clientId") Long clientId, @PathVariable("vehicle-id") Integer vehicleId) {
+
+        return Optional.of(getParkingTicketForClientVehicle(getParkingClientById(clientId), vehicleId))
+                .map(parkingTicket -> ResponseEntity.ok(new ParkingTicketResource(parkingTicket)))
+                .orElseThrow(() -> new ParkingTicketNotFoundForVehicleException("Could not find ticket for vehicle id: ", vehicleId));
+    }
+
+    @PutMapping("/{clientId}/vehicles/{vehicle-id}/parking-ticket/activate")
+    public ResponseEntity<ResourceSupport> activateParkingTicket(@PathVariable("clientId") final Long clientId, @PathVariable("vehicle-id") final Integer vehicleId) {
+
+        ParkingTicket parkingTicket = (getParkingTicketForClientVehicle(getParkingClientById(clientId), vehicleId));
+
+        if (parkingTicket.getStatus() == ParkingTicket.Status.INACTIVE) {
+            parkingTicket.setStartTime(LocalDateTime.now());
+            parkingTicket.setStatus(ParkingTicket.Status.ACTIVE);
+            return ResponseEntity.ok(new ParkingTicketResource(ticketsRepository.save(parkingTicket)));
+        } else return ResponseEntity
+                .status(HttpStatus.METHOD_NOT_ALLOWED)
+                .body(new VndErrors.VndError("Method not allowed", "You can't activate ticket that is in the " + parkingTicket.getStatus() + " status"));
+    }
+
+    @DeleteMapping("/{clientId}/vehicles/{vehicle-id}/parking-ticket/close")
+    public ResponseEntity<ResourceSupport> closeParkingTicket(@PathVariable("clientId") final Long clientId, @PathVariable("vehicle-id") final Integer vehicleId) {
+
+        ParkingClient parkingClient = getParkingClientById(clientId);
+
+        ParkingTicket parkingTicket = getParkingTicketForClientVehicle(parkingClient, vehicleId);
+
+        if (parkingTicket.getStatus() == ParkingTicket.Status.ACTIVE) {
+            parkingTicket.setEndTime(LocalDateTime.now());
+            Double parkingCost = parkingTicketsUtil.calculateParkingTicketCost(parkingTicket, parkingClient.getDriverType());
```

Comment:
Kontroler to nie jest najlepsze miejsce na taką logikę.

Reply:
Z racji, że to mały projekt dla testu, tak to uprościłem. Ale rozumiem, controller to nie miejsce na takie rzeczy.